scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -1139,7 +1139,7 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
   def termName(): Term.Name = name(Term.Name(_))
   def typeName(): Type.Name = name(Type.Name(_))
   @inline private def anonNameEmpty(): Name.Anonymous = autoPos(Name.Anonymous())
-  @inline private def anonName(): Name.Anonymous = atCurPosNext(Name.Anonymous())
+  @inline private def anonName(): Name.Anonymous = atCurPosNext(Name.Anonymous(token.name))
```

Comment:
Shouldn't we have a separate name like `Name.This` and `Name.Underscore` instead of using token.name? Since those are actual keywords?

Reply:
interesting idea. also, should we use super as well?

but underscore in self type seems different, no? in fact, underscore _is the anonymous name_, while the current usage mostly applies when there's no name 🙂

i initially wanted to use Name.Indeterminate for them but it gets escaped with backquotes.

Reply:
`Name.super` probably also? Since it is used similar to normal identifiers. You're right about `_` that is an actual anonymous name,

Reply:
added Name.This, reused Name.Placeholder since it was used for a very similar purpose.

Reply:
Looks like this broke some tests :scream:

Reply:
fix tests, or find another way?

Reply:
I would try to fix it if possible, these new names should be actually treated the same as anonymous was